name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/ComdirectPDFExtractor.java

```diff
@@ -1988,14 +2006,29 @@ private void fixMissingCurrencyConversionForSaleTaxesTransactions(Collection<Tra
         );
     }
 
-    private String concat(String first, String second)
+    /**
+     * Concatenates two strings with a specified separator.
+     * If both strings are null, the result is null.
+     * If the first string is not null and the second is null, the first string is returned.
+     * If both strings are non-null and equal, only the first string is returned.
+     * Otherwise, the two strings are concatenated with the specified separator.
+     *
+     * @param first     The first string to concatenate.
+     * @param second    The second string to concatenate.
+     * @param separator The separator to use when concatenating non-null strings.
+     * @return The concatenated string or null if both input strings are null.
```

Comment:
I appreciate the effort to add more documentation.
But I am very, very skeptical of such parameter documentation.
Do I really need the written information that ```String first``` is the "first string to concatenate"?!?
I am expecting the developer to understand and read the method signature.

In fact, personally I first read the method signature. And mostly only read the method signature.
Documentation is not compiled nor tested and can get quickly outdated. And it's not that we are writing libraries here.